MapboxNavigation/ErrorCodes.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/**
+ Enum used for indicating the type of error occured while speaking an instruction.
+ */
+public enum SpokenVoiceError: Int {
+    /**
+     Default error.
+     */
+    case defaultError
+    
+    /**
+     An intruction interrupted another instruction.
+     */
+    case overlappingInstruction
```

Comment:
Is this really an error? Unlike the other cases, one instruction interrupting another instruction can be completely expected, like when the user triggers a reroute while an instruction is being read aloud.

Reply:
We need to know when something like this happens, as it _shouldn't_. Would you rather have this filed under general?

Reply:
I think I’d rather that overlapping instructions not result in an “error” at all, but it would be fine to have a dedicated delegate method indicating that an instruction was interrupted before completion.

Reply:
> a dedicated delegate method indicating that an instruction was interrupted before completion

For example, `voiceController(_:didInterrupt:with:)`, taking two SpokenInstruction objects.